src/localplayer.cpp

```diff
@@ -726,354 +72,52 @@ v3f LocalPlayer::getEyeOffset() const
-			}
-			m_sneak_node_bb_ymax = cb_max;
-		}
-
-		/*
-			If sneaking, the player's collision box can be in air, so
-			this has to be set explicitly
-		*/
-		if (sneak_node_found && control.sneak)
-			touching_ground = true;
-	}
-
-	/*
-		Set new position but keep sneak node set
-	*/
-	bool sneak_node_exists = m_sneak_node_exists;
-	setPosition(position);
-	m_sneak_node_exists = sneak_node_exists;
-
-	/*
-		Report collisions
-	*/
-	// Dont report if flying
-	if (collision_info && !(player_settings.free_move && fly_allowed)) {
-		for (const auto &info : result.collisions) {
-			collision_info->push_back(info);
-		}
-	}
+	return m_client ? m_client->ndef() : NULL;
+}
 
-	if (!result.standing_on_object && !touching_ground_was && touching_ground) {
-		m_client->getEventManager()->put(new SimpleTriggerEvent(MtEvent::PLAYER_REGAIN_GROUND));
-		// Set camera impact value to be used for view bobbing
-		camera_impact = getSpeed().Y * -1;
-	}
+void LocalPlayer::handleAttachedMove()
+{
+	setPosition(overridePosition);
+}
 
-	{
-		camera_barely_in_ceiling = false;
-		v3s16 camera_np = floatToInt(getEyePosition(), BS);
-		MapNode n = map->getNodeNoEx(camera_np);
-		if (n.getContent() != CONTENT_IGNORE) {
-			if (nodemgr->get(n).walkable && nodemgr->get(n).solidness == 2)
-				camera_barely_in_ceiling = true;
-		}
-	}
+float LocalPlayer::getStepHeight() const
+{
+	// Player object property step height is multiplied by BS in
+	// /src/script/common/c_content.cpp and /src/content_sao.cpp
+	return (m_cao == nullptr)
+			       ? 0.0f
+			       : (touching_ground ? m_cao->getStepHeight() : (0.2f * BS));
+}
 
-	/*
-		Update the node last under the player
-	*/
-	m_old_node_below = floatToInt(position - v3f(0, BS / 2, 0), BS);
-	m_old_node_below_type = nodemgr->get(map->getNodeNoEx(m_old_node_below)).name;
+void LocalPlayer::reportRegainGround()
+{
+	m_client->getEventManager()->put(
+			new SimpleTriggerEvent(MtEvent::PLAYER_REGAIN_GROUND));
 
-	/*
-		Check properties of the node on which the player is standing
-	*/
-	const ContentFeatures &f = nodemgr->get(map->getNodeNoEx(getStandingNodePos()));
-	// Determine if jumping is possible
-	m_can_jump = touching_ground && !in_liquid;
-	if (itemgroup_get(f.groups, "disable_jump"))
-		m_can_jump = false;
-	// Jump key pressed while jumping off from a bouncy block
-	if (m_can_jump && control.jump && itemgroup_get(f.groups, "bouncy") &&
-			m_speed.Y >= -0.5 * BS) {
-		float jumpspeed = movement_speed_jump * physics_override_jump;
-		if (m_speed.Y > 1) {
-			// Reduce boost when speed already is high
-			m_speed.Y += jumpspeed / (1 + (m_speed.Y / 16 ));
-		} else {
-			m_speed.Y += jumpspeed;
-		}
-		setSpeed(m_speed);
-		m_can_jump = false;
-	}
+	// Set camera impact value to be used for view bobbing
+	camera_impact = getSpeed().Y * -1;
 }
 
-float LocalPlayer::getSlipFactor(Environment *env, const v3f &speedH)
+void LocalPlayer::calculateCameraInCeiling(Map *map, const NodeDefManager *nodemgr)
```

Comment:
the NodeDefManager commonly called `ndef` instead

Reply:
I just counted, and you're right: `ndef` wins over `nodemgr` by about 900 : 200. I'll change it.

Reply:
Changed.